### ISSUE TITLE
Demo application eats 13% CPU without doing anything...

### ISSUE THREAD

**Issue by tapika** [apetrone/simplefilewatcher#4]
Current API uses polling mechanism to poll updates, which eats CPU quite heavily.

Of course if you have 1 second polling mechanism like 
https://github.com/RuntimeCompiledCPlusPlus/RuntimeCompiledCPlusPlus/wiki
does, then CPU usage cannot be seen, as it does not consume so much CPU as in SimpleDemo, but it consumes CPU still.

Would it be difficult to recode API to use threads instead ?

Also in newest c++ standards I've noticed that new class std::filesystem appeared, which deals with file system path/folders creation / directory / files enumeration - I think it's perfect from standardization perspective. Would like to see file watcher also standardized. Does makes any sense to switch api to lower case and move into std namespace, so it would be easier to integrate later on under the same umbrella ? :-)

**Comment by morgoth990** [comment]
The SimpleDemo usa a lot of cpu because of the while loop, profiling the performance the cpu time is irrelevant:
![image](https://user-images.githubusercontent.com/40690017/56494618-1b569380-64f3-11e9-8358-6196add55b63.png)

**Comment by tapika** [comment]
yes, but cpu usage can be minimized by adding additional sleep(50 ms) in that for loop, then it won't eat CPU so much.

**Comment by lucdewit2001** [comment]
yea i agree, you dont need to watch the file as fast as possible, you can limit it to cpu usage. This is what should be done
